### PR TITLE
feat: implement new auth flow

### PR DESCRIPTION
**Pull request by botanical**
**Summary:** Summary of changes

Addresses [VEDA-134: Develop amazing new feature](https://github.com/NASA-IMPACT/veda-data-airflow/issues/134)

## Changes

* New auth flow now has user directed to Cognito login and redirected back to the API
* Added `/auth/me` endpoint

- Deployed and tested changes on veda-pipeline-dev-jtran_workflows_http_api

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

**Comment by botanical** [comment]
I noticed we have a changelog but it doesn't seem to be updated anymore. Should I add my changes to that, or have we decided to not use it? 🤔

File: workflows_api/runtime/src/auth.py [hunk 1]
```diff
@@ -18,56 +26,37 @@ def get_settings() -> config.Settings:
...
-    token: security.HTTPAuthorizationCredentials = Depends(token_scheme),
-    jwks: KeySet = Depends(get_jwks),
-) -> JWTClaims:
-    """
-    Validate & decode JWT
-    """
+def validated_token(
+    token_str: Annotated[str, Security(oauth2_scheme)],
+    required_scopes: security.SecurityScopes,
+):
```
Comment (smohiudd) [inline workflows_api/runtime/src/auth.py:32]: Can you include type hint?

Comment (botanical) [reply workflows_api/runtime/src/auth.py]: https://github.com/NASA-IMPACT/veda-data-airflow/pull/138/commits/c4f9829bb06b7c047dee8f62cc58a58dd68c0f34

**Comment by ividito** [comment]
> I noticed we have a changelog but it doesn't seem to be updated anymore. Should I add my changes to that, or have we decided to not use it? 🤔

I think this is an artifact of a time where we were experimenting with different ways to version deployed code and maintain release notes. On `veda-backend` we ended up doing this using Github releases, which is way easier than maintaining the changelog manually. It might be worth trying to crosswalk those actions into this repo to start maintaining something similar.

**Review by ividito — APPROVED** [review]

**Review by smohiudd — APPROVED** [review]

**Pull request merged**